nmtwrapper.go

```diff
@@ -0,0 +1,142 @@
+package rsmt2d
+
+// The contents of this file have been adapted from the source file available at https://github.com/celestiaorg/celestia-app/blob/main/pkg/wrapper/nmt_wrapper.go,
```

Comment:
Why copy and why not have these tests next to the wrapper?

Reply:
Or why not move the wrapper here completely then?

Reply:
> Or why not move the wrapper here completely then?

That also makes sense to me, cannot see any immediate issue for doing so, however, since it requires some refactoring in the code, I'd say it merits a separate PR. Please see the tracking issue https://github.com/celestiaorg/rsmt2d/issues/248.